generators/typescript-rxjs/npm/src/index.ts

```diff
@@ -1,70 +1,126 @@
-import { RequestBodies, RequestConversion, StandardResponse, TransformCallType, TransformRequest, RequestOpts, RequestConversions, HttpQuery, encodeURI } from '@principlestudios/openapi-codegen-typescript';
+import {
+	RequestBodies,
+	RequestConversion,
+	StandardResponse,
+	TransformCallType,
+	RequestOpts,
+	RequestConversions,
+	HttpQuery,
+	encodeURI,
+} from '@principlestudios/openapi-codegen-typescript';
 
-import { Observable, Subject, Subscriber, merge } from 'rxjs';
-import { map } from 'rxjs/operators';
-import { ajax, AjaxRequest, AjaxResponse } from 'rxjs/ajax';
+import { Observable, of } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { ajax, AjaxError, AjaxRequest, AjaxResponse } from 'rxjs/ajax';
 
-const queryString = (params: HttpQuery): string => Object.keys(params)
-    .map((key) => {
-        const value = params[key];
-        return (value instanceof Array)
-            ? value.map((val) => `${encodeURI(key)}=${encodeURI(val)}`).join('&')
-            : `${encodeURI(key)}=${encodeURI(value)}`;
-    })
-    .join('&');
+const queryString = (params: HttpQuery): string =>
```

Comment:
For the future, we should be able to just use `URLSearchParams` so that we don't have to manually build the query string